environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -425,6 +426,14 @@ private void editEnvironmentParameters(EnvironmentEditDto editDto, Environment e
                     parametersDto.getAwsParametersDto().setFreeIpaSpotPercentage(awsOriginalParameters.getFreeIpaSpotPercentage());
                     validateAwsParameters(environment, parametersDto);
                 }
+                if (originalParameters instanceof GcpParameters) {
+                    GcpParameters gcpOriginalParameters = (GcpParameters) originalParameters;
+                    String encryptionKey = gcpOriginalParameters.getEncryptionKey();
+                    ValidationResult validationResult = environmentService.getValidatorService().validateEncryptionKey(encryptionKey, editDto.getAccountId());
+                    if (validationResult.hasError()) {
+                        throw new BadRequestException(validationResult.getFormattedErrors());
+                    }
+                }
```

Comment:
Sorry, I do not get the purpose of this. The commit title suggests that the environment does not have encryption enabled, so there is no encryption key to validate in `gcpOriginalParameters`. On the other hand, the new encryption key is presumably present in `editDto` / `parametersDto`, so why not validate that one?

Reply:
My bad. Thank you pointing it out. Updated :)